geth/peers/topicpool_test.go

```diff
@@ -182,10 +183,63 @@ func (s *TopicPoolSuite) TestTheMostRecentPeerIsSelected() {
 	s.topicPool.processFoundNode(s.peer, peer2)
 	s.topicPool.processFoundNode(s.peer, peer3)
 	s.topicPool.ConfirmAdded(s.peer, discover.NodeID(peer1.ID))
+	s.topicPool.ConfirmAdded(s.peer, discover.NodeID(peer2.ID))
+	s.topicPool.ConfirmAdded(s.peer, discover.NodeID(peer3.ID))
```

Comment:
because removal logic will be executed only if peer is in the connectedPeers map, and we want it to be executed because confirmDropped adds peer to peer queue if peer was dismissed